packages/components/src/Tree/utils/treeBackgroundColor.ts

```
@@ -0,0 +1,60 @@
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import { itemSelectedColor, Theme } from '@looker/design-tokens'
+import { css } from 'styled-components'
+import { TreeBackgroundStyleProps } from '../Tree'
+
+export const treeBackgroundColor = ({
+  brand,
+  disabled,
+  hovered,
+  selected,
+  theme: { colors },
+}: TreeBackgroundStyleProps & { theme: Theme }) => {
+  const brandColors = {
+    all: colors.keySubtle,
+    hovered: colors.ui1,
+    selected: colors.keySubtle,
+  }
+  const defaultColors = {
+    all: itemSelectedColor(colors.ui2),
+    hovered: colors.ui1,
+    selected: itemSelectedColor(colors.ui2),
+  }
+  const stateColors = brand ? brandColors : defaultColors
+  let renderedColor
+
+  if (disabled) renderedColor = colors.background
+  else if (selected && hovered) renderedColor = stateColors.all
+  else if (selected) renderedColor = stateColors.selected
+  else if (hovered) renderedColor = stateColors.hovered
+  else renderedColor = colors.background
```

Aha, my bad. My earlier suggestion to add `colors.background` as the fallback is covering up the border style.

---

Ah good catch. I'm a bit stumped as to what the ideal solution is here. `colors.background` plays nice with dark backgrounds, but not with border. And "none" plays nice with border but not dark backgrounds.

Would accepting `border` as a part of the parameter object and using `colors.background` or "none" depending on if `border` is true be the right answer? The downside to this is that using border would be unsupported for dark background afaik.

---

Tagging @lukelooker and @jhardy if interested

---

@ElliotTheHuman I'd say for this PR go back to `none` like it was before. Has anyone complained about `Tree` on a dark bg? If/when they do, a new PR can figure it out.

---

Random thought – I wonder why the `yarn image-snapshots` didn't complain? It looks like on master, it isn't showing the borders: https://github.com/looker-open-source/components/blob/master/packages/components/snapshots/Tree/Border-snap.png
Perhaps the renderer can't handle the gradient...

---

Eek. That's a bummer 😞 

For future reference, I will say that one of my monitors has had trouble rendering the `border` unless I zoom in. No issues when it comes to my Macbook Pro's screen, but my external monitor appears as if it isn't rendering the `border` (or does it so faintly that the naked eye can't see it).